lib/internal/test_runner/utils.js

```diff
@@ -103,7 +102,13 @@ const kDefaultDestination = 'stdout';
 async function getReportersMap(reporters, destinations) {
   return SafePromiseAllReturnArrayLike(reporters, async (name, i) => {
     const destination = kBuiltinDestinations.get(destinations[i]) ?? createWriteStream(destinations[i]);
-    let reporter = await requireOrImport(kBuiltinReporters.get(name) ?? name);
+
+    // Load the test reporter passed to --test-reporter
+    let reporterPath = kBuiltinReporters.get(name) ?? name;
+    const { esmLoader } = require('internal/process/esm_loader');
+    const { pathToFileURL } = require('internal/url');
+    const file = isAbsolute(reporterPath) ? pathToFileURL(reporterPath).href : reporterPath;
+    let reporter = await esmLoader.import(file, undefined, { __proto__: null });
```

Comment:
I think this is a bit problematic in the way it mixes path and file: URLs, I think we should remove the "convert an absolute path to a file: URL" part, AFAIK `--import` does not do that. https://github.com/nodejs/node/blob/af0921d877fcf6b572251283da51a24c02854c8e/lib/internal/process/esm_loader.js#L92-L96
```suggestion
    const reporterSpecifier = kBuiltinReporters.get(name) ?? name;
     let cwd;
     try {
       cwd = process.cwd() + '/';
     } catch {
       cwd = '/';
     }
    const { esmLoader } = require('internal/process/esm_loader');
    let reporter = await esmLoader.import(
      reporterSpecifier,
      pathToFileURL(cwd).href,
      ObjectCreate(null),
    );
```

Reply:
This seems fine to me; my code was just the ESM part of `requireOrImport` moved here. @MoLow would these proposed changes break anything? Note the setting of the parent as CWD.

This does make me feel like we need a more proper internal API for module loading, but that can be its own PR.